packages/datetime2/src/components/date-input2/dateInput2.tsx

```diff
@@ -31,8 +31,8 @@ export interface DateInput2Props extends Omit<DateInputProps, "onChange" | "valu
     /** The default timezone selected. Defaults to the user local timezone */
     defaultTimezone?: string;
 
-    /** Callback invoked whenever the date or timezone has changed. ISO string */
-    onChange: (newDate: string, isUserChange?: boolean) => void;
+    /** Callback invoked whenever the date or timezone has changed. ISO string. Null if's not currently a valid date */
```

Comment:
I think we should update the datetime package types to reflect the `| null` possibility (that package doesn't use strict null checks yet so it hasn't been required thus far). I would also slightly tweak this language:

```suggestion
    /**
     * Callback invoked whenever the date or timezone has changed.
     *
     * @param newDate ISO string or `null` (if the date is invalid or text input has been cleared)
     * ...
     */
```